sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -93,6 +93,21 @@ private async ValueTask<bool> AuthorizeRequestOnChallengeAsyncInternal(HttpMessa
 
             string authority = GetRequestAuthority(message.Request);
             string scope = AuthorizationChallengeParser.GetChallengeParameterFromResponse(message.Response, "Bearer", "resource");
+
+            string error = AuthorizationChallengeParser.GetChallengeParameterFromResponse(message.Response, "Bearer", "error");
+            string claims = null;
+
+            if (error != null)
+            {
+                // The challenge response contained an error.
+                string base64Claims = AuthorizationChallengeParser.GetChallengeParameterFromResponse(message.Response, "Bearer", "claims");
+
+                if (error == "insufficient_claims" && base64Claims != null)
```

Comment:
Are we confident that this will be an exact match?  Any chance, for example, that some authorities may change capitalization?

Reply:
Yes; [the docs](https://learn.microsoft.com/entra/identity-platform/claims-challenge?tabs=dotnet#claims-challenge-header-format) specify `insufficient_claims`, and in practice there's also only one server-side implementation.

Reply:
According to [this documentation](https://learn.microsoft.com/en-us/entra/identity-platform/claims-challenge?tabs=dotnet#claims-challenge-header-format), for this scenario the error will always be "insufficient_claims", but I can lowercase the error content just to be sure.

Reply:
I wouldn't. It's a waste of time/memory and our de/serialization story is already intentionally (by default) case-sensitive because that's what we document. This is congruent. Architects consider it a bug in either client or server code if it returns a different case than what is documented.